tools/integration_tests/gzip/gzip_test.go

```diff
@@ -172,9 +183,62 @@ func destroy_testdata(m *testing.M) error {
 	return nil
 }
 
+// createContentOfSize generates a string of the specified content size in bytes.
+// Failure cases:
+// 1. contentSize <= 0
+func createContentOfSize(contentSize int) (string, error) {
+	// fail if contentSize <= 0
+	if contentSize <= 0 {
+		return "", fmt.Errorf("unsupported fileSize: %d", contentSize)
+	}
+
+	// Create text-content of given size.
+	// strings.builder is used as opposed to string appends
+	// as this is much more efficient when multiple concatenations
+	// are required.
+	var contentBuilder strings.Builder
```

Comment:
nit: you know the size of the string that will be stored in contentBuilder. So, you can call contentBuilder.Grow(contentSize) to pre-allocate the necessary space.